src/Illuminate/Auth/Guard.php

```diff
@@ -375,11 +375,12 @@ public function attempt(array $credentials = [], $remember = false, $login = tru
 		// fact valid we'll log the users into the application and return true.
 		if ($this->hasValidCredentials($user, $credentials))
 		{
-			if ($login) $this->login($user, $remember);
+			if ($login)
 			{
-				return true;
-
+				$this->login($user, $remember);
```

Comment:
"return true" inside into the braces and below "$this->login($user, $remember);

Otherwise its always true.